Warning in strncpy rmw_dds_common

I have found this warning on rmw_dds_common https://ci.ros2.org/job/ci_windows/9763/console

```
17:03:59 C:\ci\ws\src\ros2\rmw_dds_common\rmw_dds_common\test\test_graph_cache.cpp(195,8): warning C4996: 'strncpy': This function or variable may be unsafe. Consider using strncpy_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [C:\ci\ws\build\rmw_dds_common\test_graph_cache.vcxproj]

17:03:59 C:\ci\ws\src\ros2\rmw_dds_common\rmw_dds_common\test\test_graph_cache.cpp(388,8): warning C4996: 'strncpy': This function or variable may be unsafe. Consider using strncpy_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [C:\ci\ws\build\rmw_dds_common\test_graph_cache.vcxproj]
```

--- a/rmw_dds_common/test/test_graph_cache.cpp
+++ b/rmw_dds_common/test/test_graph_cache.cpp
@@ -192,7 +192,7 @@ gid_from_string(const std::string & str)
 {
   rmw_gid_t gid = {};
   EXPECT_LT(str.size(), RMW_GID_STORAGE_SIZE);
-  strncpy(
+  memcpy(
     reinterpret_cast<char *>(gid.data),
     str.c_str(),
     RMW_GID_STORAGE_SIZE);
@@ -385,7 +385,7 @@ gid_msg_from_string(const std::string & str)
   rmw_dds_common::msg::Gid gid;
   gid.data = {};
   EXPECT_LT(str.size(), RMW_GID_STORAGE_SIZE);
-  strncpy(
+  memcpy(
     reinterpret_cast<char *>(gid.data.data()),
     str.c_str(),
     RMW_GID_STORAGE_SIZE);

--- a/rmw_dds_common/test/test_graph_cache.cpp
+++ b/rmw_dds_common/test/test_graph_cache.cpp
@@ -14,8 +14,8 @@
 
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
+#include <string.h>
 
-#include <cstring>
 #include <string>
 #include <tuple>
 #include <utility>
@@ -192,7 +192,7 @@ gid_from_string(const std::string & str)
 {
   rmw_gid_t gid = {};
   EXPECT_LT(str.size(), RMW_GID_STORAGE_SIZE);
-  std::strncpy(
+  strncpy(
     reinterpret_cast<char *>(gid.data),
     str.c_str(),
     RMW_GID_STORAGE_SIZE);
@@ -385,7 +385,7 @@ gid_msg_from_string(const std::string & str)
   rmw_dds_common::msg::Gid gid;
   gid.data = {};
   EXPECT_LT(str.size(), RMW_GID_STORAGE_SIZE);
-  std::strncpy(
+  strncpy(
     reinterpret_cast<char *>(gid.data.data()),
     str.c_str(),
     RMW_GID_STORAGE_SIZE);

--- a/rmw_dds_common/test/test_graph_cache.cpp
+++ b/rmw_dds_common/test/test_graph_cache.cpp
@@ -195,7 +195,7 @@ gid_from_string(const std::string & str)
   memcpy(
     reinterpret_cast<char *>(gid.data),
     str.c_str(),
-    RMW_GID_STORAGE_SIZE);
+    str.size() + 1);
   return gid;
 }
 
@@ -388,7 +388,7 @@ gid_msg_from_string(const std::string & str)
   memcpy(
     reinterpret_cast<char *>(gid.data.data()),
     str.c_str(),
-    RMW_GID_STORAGE_SIZE);
+    str.size() + 1);
   return gid;
 }
 

